configsuite/schema.py

```diff
@@ -32,13 +32,28 @@ def _context_basic_only(elem):
     )
 
 
+@configsuite.validator_msg("Is default value of correct type")
+def _correct_default_type(elem):
+    if MK.Default not in elem:
+        return True
+    # If Context validator not applied
+    # If Context Transformator, no validation at all
+    # Else:
+    # - apply layer transform
+    # - apply transform
+    # - validate type
+    # - apply validators
+    return bool(elem[MK.Type].validate(elem[MK.Default]))
+
+
 _META_SCHEMA = {
     MK.Type: types.NamedDict,
-    MK.ElementValidators: (_context_basic_only,),
+    MK.ElementValidators: (_context_basic_only, _correct_default_type),
```

Comment:
This is more tricky then I first expected. I suggest that we leave this for now (removing the entire `_correct_default_type`) and then, if we are to have a look at this we should do it for all layers, not only the default values in the schema.

Reply:
Sure, I think this PR will have plenty enough to handle on its own